### DISCUSSION TITLE
Not routing via channel layer

### DISCUSSION THREAD

**Discussion by 4f3x** [django/channels#1715] [Help!] [answered]
Hello,

I am trying to route MQTT through Channels. However, when fetching the `channel_layer`, `send`ing does not appear to have any effect. Here is the setup:

```python
# consumers.py
class MqttConsumer(SyncConsumer):
    def __init__(self): # this constructor is never called ...
        super().__init__()
        print(self.channel_layer)
        print(self.channel_name)

    def mqtt_msg(self, event):
        print(event)
```

```python
# routing.py
application = ProtocolTypeRouter({
    'mqtt': MqttConsumer
})
```

```python
# send.py
async def send_message():
    await channel_layer.send('mqtt', { 'type': 'mqtt.msg', 'text': 'test' })
```

I have tried both `InMemoryChannelLayer` and `RedisChannelLayer` backends to no effect. What appears to be the issue? How can I provide further logs, as I have no error messages/exceptions at all. Thanks a lot in advance.

**Comment by 4f3x** [comment]
My issue was closed without any feedback. How am I to provide helpful information to resolve this, seeing as no error messages are produced at all? The documentation of this project is sparse at best on how to implement MQTT, even though it says so on the introductory page.

**Reply by carltongibson** [reply]
It was moved to a discussion. It's not an issue. It's a question.

**Reply by 4f3x** [reply]
Would you kindly instruct me on how I can provide further information to resolve my issue? Cheers.

**Comment by 4f3x** [comment, marked as answer]
Got it to work. `asgiref` needed to be downgraded, the worker didn't instantiate properly.
